src/pages/UseCoins/UseCoins.js

```diff
@@ -1,10 +1,38 @@
 import { useHistory } from 'react-router-dom'
+import DistributionButton from '../../components/DistributionButton/DistributionButton'
+import { Link } from 'react-router-dom'
 
-export default function UseCoins() {
+const transactionTypes = [
+  {
+    type: 'grow',
+    color: 'blue',
+  },
+  {
+    type: 'spend',
+    color: 'orange',
+  },
+  {
+    type: 'share',
+    color: 'purple',
+  },
+]
+
+export default function UseCoins({ spendCoins, lastEarning }) {
   const history = useHistory()
   return (
     <div>
-      <button onClick={() => history.push('/home')}>Return</button>
+      {transactionTypes.map(({ type, color }) => {
+        return (
+          <DistributionButton
+            key={type}
+            type={type}
+            onAdd={() => spendCoins(type, -1)}
```

Comment:
You simply flipped the signs, I think this is confusing interface wise if the `+` button reduces the number. Maybe `DistributionButton` is not the right component?